### PR TITLE
Firebase Database fixes for startAt and endAt

### PR DESCRIPTION
**Pull request by collinjackson**
This prevents exceptions when startAt/endAt is used without a key on iOS, and also allows values that don't fit in a Java int on Android.

Fixes flutter/flutter#10899
Fixes flutter/flutter#10900

### CODE REVIEW

File: packages/firebase_database/android/src/main/java/io/flutter/plugins/firebase/database/FirebaseDatabasePlugin.java [hunk 1]
```diff
@@ -69,28 +69,52 @@ private Query getQuery(Map<String, Object> arguments) {
...
-        query = query.endAt((Double) startAt);
-      } else if (startAt instanceof Integer) {
-        query = query.startAt((int) startAt);
+      if (parameters.containsKey("startAtKey")) {
+        String startAtKey = (String) parameters.get("startAtKey");
+        if (startAt instanceof Boolean) {
+          query = query.startAt((Boolean) startAt, startAtKey);
+        } else if (startAt instanceof String) {
+          query = query.startAt((String) startAt, startAtKey);
+        } else if (startAt instanceof Integer) {
```
Comment (jason-simmons) [inline packages/firebase_database/android/src/main/java/io/flutter/plugins/firebase/database/FirebaseDatabasePlugin.java]: The cases for the Integer type can be removed and ints can be processed like other numbers.  It looks like Firebase's Query only handles double/string/bool values and doesn't have any special handling for ints.

File: packages/firebase_database/ios/Classes/FirebaseDatabasePlugin.m [hunk 2]
```diff
@@ -40,11 +40,21 @@ - (FlutterError *)flutterError {
...
+    } else {
+      query = [query queryStartingAtValue:startAt];
+    }
   }
   id endAt = parameters[@"endAt"];
   if (endAt) {
-    query = [query queryEndingAtValue:endAt childKey:parameters[@"endAtKey"]];
+    id endAtKey = parameters[@"endAtKey"];
+    if (endAtKey) {
+      query = [query queryEndingAtValue:endAt childKey:parameters[@"endAtKey"]];
```
Comment (jason-simmons) [inline packages/firebase_database/ios/Classes/FirebaseDatabasePlugin.m]: just use childKey:endAtKey here

File: packages/firebase_database/android/src/main/java/io/flutter/plugins/firebase/database/FirebaseDatabasePlugin.java [hunk 3]
```diff
@@ -69,28 +69,48 @@ private Query getQuery(Map<String, Object> arguments) {
...
-        query = query.endAt((Double) endAt);
-      } else if (endAt instanceof Integer) {
-        query = query.endAt((int) endAt);
+      if (parameters.containsKey("endAtKey")) {
+        String endAtKey = (String) parameters.get("endAtKey");
+        if (endAt instanceof Boolean) {
+          query = query.endAt((Boolean) endAt, endAtKey);
+        } else if (endAt instanceof String) {
+          query = query.endAt((String) endAt, endAtKey);
+        } else if (endAt instanceof Integer) {
```
Comment (jason-simmons) [inline packages/firebase_database/android/src/main/java/io/flutter/plugins/firebase/database/FirebaseDatabasePlugin.java]: also remove this in the endAt and equalTo filters

Comment (collinjackson) [inline packages/firebase_database/android/src/main/java/io/flutter/plugins/firebase/database/FirebaseDatabasePlugin.java]: done

**Pull request merged**